unleash.go

```diff
@@ -44,6 +44,12 @@ func Initialize(options ...ConfigOption) (err error) {
 	return
 }
 
+// IsReady is a channel operated method which specifies when the global unleash
+// client is ready
+func IsReady() bool {
```

Comment:
@shividhar This is a perfectly valid use case, but I don't think this solution will work unfortunately. Since the function reads from a channel, it will block the calling routine until the channel is written to. I was thinking the quick fix would be that you simply rename the function to `WaitForReady()` which will clearly indicate that it will block until the Unleash client is ready. However, calling this function a second time (your use case) would then hang indefinitely because the channel is now empty and only written to once. Also if you register a `RepositoryListener`, then it will create a race condition with either the listener _or_ this `WaitForReady` receiving the value and the loser would then hang forever. Obviously this is suboptimal and can lead to non-intuitive use of the library.

This is a very good use case though and exposes a limitation in the Go implementation of the client that doesn't exist in the Javascript version because EventEmitters can have multiple listeners. I'm going to give this some more thought and see if I can come up with an elegant solution in the library. Feel free to do the same by updating this pull request if you have some ideas.

In the meantime, I think a good workaround for this would be making a struct that implements the `RepositoryListener` interface (`OnReady`) and sets a boolean to track that it has been initialized. You could then make your own version of `IsEnabled` that waits for the signal before calling the default client. Something like this:

```go
type UnleashWrapper struct {
	isReady bool
	onReady chan bool
}

func (h *UnleashWrapper) IsEnabled(feature string, options ...unleash.FeatureOption) bool {
	if !h.isReady {
		h.isReady = <-h.onReady // block until ready
	}
	return unleash.IsEnabled(feature, options)
}

func (h *UnleashWrapper) OnReady() {
	h.onReady <- true
}

var wrapper UnleashWrapper

func init() {
	wrapper = UnleashWrapper{onReady: make(chan bool, 1)}
	unleash.Initialize(
		unleash.WithListener(&wrapper),
		unleash.WithAppName("my-application"),
		unleash.WithUrl("http://unleash.herokuapp.com/api/"),
	)

	wrapper.IsEnabled("foo")
}
...
```